nvbench/device_info.cu

```diff
@@ -38,8 +43,114 @@ device_info::memory_info device_info::get_global_memory_usage() const
 device_info::device_info(int id)
     : m_id{id}
     , m_prop{}
+    , m_nvml_device(nullptr)
 {
   NVBENCH_CUDA_CALL(cudaGetDeviceProperties(&m_prop, m_id));
+
+#ifdef NVBENCH_HAS_NVML
+  if (!nvml_inited)
+  {
+    NVBENCH_NVML_CALL_NO_API(nvmlInit());
```

Comment:
The documentation states that we have to call `nvmlShutdown` per each `nvmlInit` call. Are there some notes that say otherwise?

Reply:
If there happen to be a need for `nvmlShutdown`, we could wrap it into an RAII wrapper and implement it as a singleton pattern instead of a global bool.

Reply:
Great idea, I'll do this.

Reply:
Added this to a new `nvbench/internal/nvml.cxx` file 👍